en/components/hierarchicalgrid/load-on-demand.md

```diff
@@ -8,292 +8,206 @@ _keywords: Ignite UI for Angular, UI controls, Angular widgets, web widgets, UI
-            qS += `${dataState.key}?`;
-
-            if (!dataState.rootLevel) {
-                if (typeof dataState.parentID === "string") {
-                    qS += `$filter=${dataState.parentKey} eq '${dataState.parentID}'`;
-                } else {
-                    qS += `$filter=${dataState.parentKey} eq ${dataState.parentID}`;
-                }
+### Advanced filtering
+
+In order to use Advanced Filtering in the `IgxHierarchicalGrid` with load on demand, you need to set the [`schema`]({environment:angularApiUrl}/classes/igxhierarchicalgridcomponent.html#schema) property of the grid to an entity with hierarchical structure, specifying child entities and fields with their data types. This ensures that filtering expressions with nested queries can be created even before any child grid data is loaded and that the grid can correctly interpret and apply these filters to the data.
+
+In our case, this is the correct hierarchical structure:
+
+```TypeScript
+public schema: EntityType[] = [
+    {
+        name: 'Customers',
+        fields: [
+            { field: 'customerId', dataType: 'string' },
+            { field: 'companyName', dataType: 'string' },
+            { field: 'contactName', dataType: 'string' },
+            { field: 'contactTitle', dataType: 'string' }
+        ],
+        childEntities: [
+            {
+                name: 'Orders',
+                fields: [
+                    { field: 'customerId', dataType: 'string' },
+                    { field: 'orderId', dataType: 'number' },
+                    { field: 'employeeId', dataType: 'number' },
+                    { field: 'shipVia', dataType: 'string' },
+                    { field: 'freight', dataType: 'number' }
+                ],
+                childEntities: [
+                    {
+                        name: 'Details',
+                        fields: [
+                            { field: 'orderId', dataType: 'number' },
+                            { field: 'productId', dataType: 'number' },
+                            { field: 'unitPrice', dataType: 'number' },
+                            { field: 'quantity', dataType: 'number' },
+                            { field: 'discount', dataType: 'number' }
+                        ]
+                    }
+                ]
             }
-        }
-        return `${this.url}${qS}`;
+        ]
     }
-}
+];
 ```
 
-### Hierarchical Grid Setup
+#### Setting initial filter
 
-Next we will setup our hierarchical grid and connect it to our remote service provider.
+Now let's add initial filtering rules to our grid so that the root grid is filtered when first loaded. We will create [`FilteringExpressionsTree`]({environment:angularApiUrl}/classes/filteringexpressionstree.html) and set it to the [`advancedFilteringExpressionsTree`]({environment:angularApiUrl}/classes/@@igTypeDoc.html#advancedFilteringExpressionsTree) property of the `IgxHierarchicalGrid` using the `ngOnInit` lifecycle hook.
```

Comment:
Since this is a hgrid specific topic, the link ({environment:angularApiUrl}/classes/@@igTypeDoc.html#advancedFilteringExpressionsTree) does not work because @@igTypeDoc is not processed. It needs to be `igxhierarchicalgridcomponent`.